### PR TITLE
Dbus

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **dbus/1.12.20**

this are minor tweaks on top of #2147

closes #2147

let's try FreeBSD too

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
Call for reviews !
@Croydon 
@madebr 
@prince-chrismc
@uilianries
@SSE4
@danimtb

File: recipes/dbus/1.x.x/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    _cmake = None
+
+    def configure(self):
+        if self.settings.os not in ("Linux", "FreeBSD"):
+            raise ConanInvalidConfiguration("dbus is only supported on Linux")
```
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:44]:
```suggestion
            raise ConanInvalidConfiguration("dbus is only supported on Linux and FreeBSD")
```

File: recipes/dbus/1.x.x/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    description = "D-Bus is a simple system for interprocess communication and coordination."
+    topics = ("conan", "dbus")
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+        "with_x11": [True, False],
+        "with_glib": [True, False],
+        "enable_assert": [True, False],
+        "enable_checks": [True, False],
+        "install_system_libs": [True, False]}
```
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:21]: What system libs are installed?
Would it be useful to install them anyways (and don't have this option)?

File: recipes/dbus/1.x.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                                  build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        dbus_cmake = tools.os.path.join(
+            self._source_subfolder, "cmake", "CMakeLists.txt")
+
+        if self.options.with_glib:
+            tools.replace_in_file(dbus_cmake, "GLib2", "glib")
+            tools.replace_in_file(dbus_cmake, "GLIB2", "GLIB")
```
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:88]: I think this change can be made unconditionally.

File: recipes/dbus/1.x.x/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses",
+                  src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        for i in ["var", "share", "etc"]:
+            shutil.move(os.path.join(self.package_folder, i), os.path.join(self.package_folder, "res", i))
```
Comment (madebr) [inline recipes/dbus/1.x.x/conanfile.py:100]: Can setting `CMAKE_INSTALL_DATAROOTDIR` to `res` in `_configure_cmake`, avoid this move?

Comment (ericLemanissier) [reply recipes/dbus/1.x.x/conanfile.py:100]: no: https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12.20/cmake/bus/CMakeLists.txt#L103-104

Comment (ericLemanissier) [reply recipes/dbus/1.x.x/conanfile.py:100]: it's used only in https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12.20/cmake/modules/CPackInstallConfig.cmake so I'm removing the option

File: recipes/dbus/1.x.x/conanfile.py (hunk 2 above)
Comment (ericLemanissier) [reply recipes/dbus/1.x.x/conanfile.py:21]: it's used only in https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12.20/cmake/modules/CPackInstallConfig.cmake so I'm removing the option

**Review by madebr — APPROVED** [review]

File: recipes/dbus/1.x.x/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        if self.settings.os not in ("Linux", "FreeBSD"):
+            raise ConanInvalidConfiguration("dbus is only supported on Linux and FreeBSD")
+
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
```
Comment (prince-chrismc) [inline recipes/dbus/1.x.x/conanfile.py:50]: Theres a new `tools.rename` not when should start using that

File: recipes/dbus/1.x.x/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+            self._cmake.definitions["DBUS_ENABLE_XML_DOCS"] = False
+
+            self._cmake.definitions["DBUS_BUILD_X11"] = self.options.with_x11
+            self._cmake.definitions["DBUS_WITH_GLIB"] = self.options.with_glib
+            self._cmake.definitions["DBUS_DISABLE_ASSERT"] = not self.options.enable_assert
+            self._cmake.definitions["DBUS_DISABLE_CHECKS"] = not self.options.enable_checks
+
+            path_to_cmake_lists = os.path.join(self._source_subfolder, "cmake")
+
+            self._cmake.configure(source_folder=path_to_cmake_lists,
```
Comment (prince-chrismc) [inline recipes/dbus/1.x.x/conanfile.py:75]: Does this not by pass the cmake wrapper?

**Review by prince-chrismc — COMMENTED** [review]
I'll approve if you can answer the cmake wrapper question

Please also link this to close the other PR

File: recipes/dbus/1.x.x/conanfile.py (hunk 6 above)
Comment (uilianries) [reply recipes/dbus/1.x.x/conanfile.py:75]: Indeed.

Comment (ericLemanissier) [reply recipes/dbus/1.x.x/conanfile.py:75]: It bypasses it. I made several attempt to actually use the wrapper file, but this project does not seem to be ready to be used as a sub-project :/

Comment (madebr) [reply recipes/dbus/1.x.x/conanfile.py:75]: Until conan has good toolchain support, shouldn't the main cmake script then be patched to call conan_basic_setup?
On Linux, I think it's needed for libstdc++ vs libstdc++11 and other libcxx libraries.

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Pull request merged**